tutorials/roostats/rs301_splot.py

```diff
@@ -0,0 +1,318 @@
+
+    print(f"\n\nThe dataset after creating sWeights:\n")
+    data.Print()
+
+    # Check that our weights have the desired properties
+
+    print("\n\n------------------------------------------\n\nCheck SWeights:")
+    print("Yield of Z is\t", zYield.getVal(), ".  From sWeights it is ")
+    print(sData.GetYieldFromSWeight("zYield"))
+
+    print("Yield of QCD is\t", qcdYield.getVal(), ".  From sWeights it is : ")
+    print(sData.GetYieldFromSWeight("qcdYield"))
+
+    for i in range(10):
+        print("Weight for event: ", i, sData.GetSWeight(i, "zYield"))
+        print("qcd Weight: ", sData.GetSWeight(i, "qcdYield"))  
+        print("Total Weight: ", sData.GetSumOfEventSWeight(i))
+
+    # import this new dataset with sWeights
+    object.Import(data, Rename="dataWithSWeights")
+
+    ROOT.RooMsgService.instance().setGlobalKillBelow(ROOT.RooFit.INFO)
+
+
+def MakePlots(object):
+
+    # Here we make plots of the discriminating variable (invMass) after the fit
+    # and of the control variable (isolation) after unfolding with sPlot.
+    # make our canvas
+    cdata = ROOT.TCanvas("sPlot", "sPlot demo", 400, 600)
+    cdata.Divide(1, 3)
+
+    # get what we need out of the workspace
+    model = object.pdf("model")
+    zModel = object.pdf("zModel")
+    qcdModel = object.pdf("qcdModel")
+
+    isolation = object.var("isolation")
+    invMass = object.var("invMass")
+
+    # note, we get the dataset with sWeights
+    data = object.data("dataWithSWeights")
+
+    # create weighted data sets
+    # """
+    ##info for debuggin :
+    ## ROOT.nullptr doesn't work for : const ROOT.RooFormulaVar & cutvar
+    ##                                 const char* wgtVarName = nullptr
+    ##                                 const char* cuts = nullptr
+    ## not even using hex(id(nullptr))
+    ## improve pythonzation...
+    ## ROOT.RooDataSet::RooDataSet(RooStringView name, ROOT.RooStringView title, ROOT.RooDataSet* data, const ROOT.RooArgSet& vars, const ROOT.RooFormulaVar&     cutVar, const char* wgtVarName = nullptr)
+    ## ROOT.RooDataSet::RooDataSet(RooStringView name, ROOT.RooStringView title, ROOT.RooDataSet* data, const ROOT.RooArgSet& vars, const char* cuts = nu    llptr, const char* wgtVarName = nullptr)
+    # """
+    cutVar0 = ROOT.RooFormulaVar()  # continues at ...
+    # for the purpose of this tutorial we don't use cuts.
+    # however, it is possible to implement them as shown below:
+    zYield = object.var("zYield")
+    qcdYield = object.var("qcdYield")
+    ral = ROOT.RooArgList(zYield, qcdYield)
+    cutVar = ROOT.RooFormulaVar("zYield < 0 and qcdYield < 0.5", "Cuts", " @0 < 0 && @1 < 0.5", ral, True)
+    # ... continues here
+    dataw_qcd = ROOT.RooDataSet(data.GetName(), data.GetTitle(), data, data.get(), cutVar0, "qcdYield_sw")
+    dataw_z = ROOT.RooDataSet(data.GetName(), data.GetTitle(), data, data.get(), cutVar0, "zYield_sw")
```

Comment:
```suggestion
    # create weighted data sets
    dataw_qcd = ROOT.RooDataSet(data.GetName(), data.GetTitle(), data, data.get(), "", "qcdYield_sw")
    dataw_z = ROOT.RooDataSet(data.GetName(), data.GetTitle(), data, data.get(), "", "zYield_sw")
```
All of this is not necessary (see the original C tutorial). It should just take no selection cut, which can be signified by either `nullptr`, or an empty string. The latter is more convenient in Python.